overviews/core/_posts/2012-09-20-futures.md

```diff
@@ -615,8 +615,42 @@ Invoking the `future` construct uses a global execution context to start an asyn
 Support for extending the Futures API with additional utility methods is planned.
 This will allow external frameworks to provide more specialized utilities.
 
+
 ## Blocking
 
+Futures are generally asynchronous and do not block the underlying execution threads.
+However, in certain cases, it is necessary to block.
+We distinguish two forms of blocking the execution thread:
+invoking arbitrary code that blocks the thread from within the future,
+and blocking from outside another future, waiting until that future gets completed.
+
+
+### Blocking inside a Future
+
+As seen with the global `ExecutionContext`, it is possible to notify an `ExecutionContext` of a blocking call with the `blocking` construct.
+The implementation is however at the complete discretion of the `ExecutionContext`. While some `ExecutionContext` such as `ExecutionContext.global`
+implement `blocking` by means of a `ManagedBlocker`, some execution contexts such as the fixed thread pool: 
+
+    ExecutionContext.fromExecutor(Executors.newFixedThreadPool(x))
+
+will do nothing, as shown in the following:
+
+    implicit val ec = ExecutionContext.fromExecutor(
+                        Executors.newFixedThreadPool(4))
+    Future {
+      // here blocking serves only for documentation purpose
```

Comment:
@backuitist 

>You're right, the library should not instantiate its own EC, but that's not what I said. Libraries needing a blocking EC can merely declare it as a dependency, and if you want to leverage implicits you can do something along those lines:
>[...]
>As a matter of fact I believe that, as a general rule, implicit parameters are fine as long as they have a very specific meaning. blocking feels like a hack to work around the fact that we only have one kind of ExecutionContext, not 2.
But that's just my opinion, and for now I'm just suggesting that we make it clear that blocking might not work as expected under certain conditions.

I can understand that argument, I think we discussed having multiple ones initially but came to the conclusion that it was disruptive enough to have the ripple-effect of having to add `(implicit ec: ExecutionContext)` up the call chain as soon as something async needed to be done down the call chain. Adding a new dimension where one also needed to add yet another implicit as soon as blocking was involved would have serious impact on binary and source compatibility. (Imagine having to add a blocking call, now all your signatures would change, so people would just do unmanaged blocking instead, which would mean that they'd block on the provided EC, without any means of parrying for it).

My long-standing plan has been to create a thread pool that would work well for both CPU-bound work á la FJP but also blocking work á la TPE, transitioning workers between those modes based on `blocking`/`BlockContext`. What would be fantastic about that is that code that uses `blocking` would simply work with the new EC without any modification.

>I took a quick look. That's nice, but that's not yet in the scala-library and, as far as I understand, that only works if you create your TPE-based EC through your ThreadPoolConfig. If someone has a TPE at hand (provided by some other code) and wants to wrap it in an EC through fromExecutor then you're stuck.

Of course, it is not a virus that infects other's object instances :)